var/spack/repos/builtin/packages/py-esmpy/package.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyEsmpy(PythonPackage):
+    """ESMPy is a Python interface to the Earth System Modeling Framework (ESMF)"""
+
+    homepage = "https://earthsystemmodeling.org/"
+    git = "https://github.com/esmf-org/esmf.git"
+
+    # because it relies on py-setuptools-git-versioning, we need to install from git
+    version("develop", branch="develop")
+    version("8.4.0", tag="v8.4.0")
+    version("8.3.1", tag="v8.3.1")
+    version("8.3.0", tag="v8.3.0")
+    version("8.2.0", tag="v8.2.0")
+    version("8.1.1", tag="v8.1.1")
+    version("8.0.1", tag="v8.0.1")
+    version("8.0.0", tag="v8.0.0")
```

Comment:
URLs are greatly preferred over git checkouts whenever possible for security reasons. Git tags can be moved after the fact, and can't be checksummed. But if git is required to use setuptools-git-versioning, then I guess there isn't much you can do.

Reply:
I'll have a bit of a deeper dig to see if I can make the tarball checkouts build. If they don't, it seems like an upstream bug that should be fixed anyway!

Out of curiosity, what are the security reasons? Would it be safer to also specify the commit hash associated with the tags? That seems to me to be fairly similar to a tarball + sha256.

Reply:
Tarball + sha256 is safest because it's easy to verify the integrity of the tarball before extracting it. Commit hash isn't as good because when Spack downloads something from git it can verify it the first time, but it can't be verified after it's been archived in a tarball. We strip the .git directory for some reason, so there's no way to verify it even after extracting. Tag is the least safe since it can't be verified during the download or during usage. @tgamblin know more about this than I do though.